sound/soc/soc-topology.c

```diff
@@ -1478,8 +1478,7 @@ static int soc_tplg_dapm_widget_create(struct soc_tplg *tplg,
 	}
 
 	control_hdr = (struct snd_soc_tplg_ctl_hdr *)tplg->pos;
-	dev_dbg(tplg->dev, "ASoC: template %s has %d controls of type %x\n",
-		w->name, w->num_kcontrols, control_hdr->type);
+	dev_dbg(tplg->dev, "ASoC: template %s has %d controls\n", w->name, w->num_kcontrols);
```

Comment:
Rather than suppress all information on type, can we instead loop to show what types are enabled for that control?

Reply:
But then we shall have:
```
for (i = 0; i < w->num_kcontrols; i++) {
	dev_dbg();
	// or
	sprintf(dbg_buff, "");
}
```
I would rather leave that out, but I can also just drop this patch, it just occurred to me that this is now wrong.

Reply:
you dont need the loop, but maybe we can move the dev_dbg() to after creating the controls and use mixer_coutn/enum_count/bytes_count to print instead?

Reply:
@ranj063, you are right, I have moved it after the loop.